### DISCUSSION TITLE
Vitest 2

### DISCUSSION THREAD

**Discussion by sheremet-va** [vitest-dev/vitest#5210] [Feedback]
We wanted to start gathering feedback from users for the next major version.

Feel free to bring back proposals if you want to expand on them. If possible, create an issue for your feature and link it to this discussion so we can better track the proposal.

# Timeline

We did not follow any patterns for releases before. We usually try to have a breaking change release when we feel that enough time has passed since the last big breaking change and we are sure it won't disturb the ecosystem too much. We now also have an [ecosystem-ci](https://github.com/vitest-dev/vitest-ecosystem-ci) to see if other projects are ready for the release.

We will have a 1-2 month beta period before releasing a new major version. You can follow the state of PRs and issues for the next major here: https://github.com/vitest-dev/vitest/milestone/15

# Initial thoughts

We are planning to change the default pool form `threads` to `forks`:

- https://github.com/vitest-dev/vitest/pull/5047

This means that the test will spawn `child_processes` by default instead of `new Worker`. This should bring more stability with a slight performance cost.

Vitest 2 will also switch to chai 5 which is now bundled as ESM:

- https://github.com/vitest-dev/vitest/pull/5135

**Comment by ghiscoding** [comment]
Can we have a rough idea of when the Beta will start? It's not really obvious when v2 is expected and also if I understand correctly you guys will not follow Vite's version roadmap, is that right?

**Reply by sheremet-va** [reply]
Vite's roadmap follows Node EOL which we plan to follow, so in the future, they should line up. But currently, I think, we will release Vitest 2 before the next Vite major to improve stability with #5047.

**Comment by motss** [comment]
How about an improved experimental browser mode in v2 with access to Playwright APIs like simulating keyboard events, taking screenshots, etc?

**Reply by sheremet-va** [reply]
I am working on it here: https://github.com/vitest-dev/vitest/pull/5097

It should be released before Vitest 2.0

**Reply by sheremet-va** [reply]
In the end, it will be part of Vitest 2.0: https://github.com/vitest-dev/vitest/pull/5097

**Comment by sheremet-va** [comment]
The beta has been released for some time now: https://github.com/vitest-dev/vitest/releases

The current idea is to release Vitest 2.0 in June.

**Reply by charlieforward9** [reply]
Is a June release still on the table or should we expect a bit of a longer wait?

**Reply by sheremet-va** [reply]
Looks like it isn't after all 😄 I am planning to have a release this week.

**Reply by RobinTail** [reply]
Happy Sunday, @sheremet-va ! 
What a lovely sunny day to release Vitest 2.0 😸

**Reply by RobinTail** [reply]
In my personal competition between Vitest 2, Jest 30, `typescript-eslint` 8 and Express 5 — Vitest is a winner!
Thank you so much, @sheremet-va. 
Legendary!

**Comment by sheremet-va** [comment]
Vitest 2.0 is out! https://github.com/vitest-dev/vitest/releases/tag/v2.0.0
